siad/server.go

```diff
@@ -382,6 +383,23 @@ func NewServer(bindAddr, requiredUserAgent, requiredPassword string) (*Server, e
 		listener: l,
 		httpServer: &http.Server{
 			Handler: mux,
+
+			// set reasonable timeout windows for requests, to prevent the Sia API
+			// server from leaking file descriptors due to slow, disappearing, or
+			// unreliable API clients.
+
+			// ReadTimeout defines the maximum amount of time allowed to fully read
+			// the request body. This timeout is applied to every handler in the
+			// server.
+			ReadTimeout: time.Minute,
```

Comment:
gotcha. I'd still like to see a shorter `IdleTimeout` though, probably <30 seconds